src/plugins/controls/public/options_list/embeddable/options_list_embeddable.tsx

```diff
@@ -224,12 +228,26 @@ export class OptionsListEmbeddable extends Embeddable<OptionsListEmbeddableInput
               invalidSelections: newInvalidSelections,
             });
           }
-          const newFilters = await this.buildFilter();
-          this.dispatch.publishFilters(newFilters);
+          this.unpublishedChanges.next({ selectedOptions, existsSelected });
         })
     );
+
+    /**
+     * debounce filter publications for a split second for a smoother selection/deselection user experience
+     */
+    this.subscriptions.add(
+      this.unpublishedChanges.pipe(debounceTime(750)).subscribe(() => {
+        this.publishNewSelections();
+      })
+    );
```

Comment:
Since we are flushing out unpublished changes with `this.unpublishedChanges.next({});`, we need to prevent these events from firing the `publishNewSelections` method - otherwise, you end up in an infinite loop where you are endlessly publishing changes. Put a print statement in `publishNewSelections()` and you'll see that it continuously fires ever `750ms` as soon as the dashboard loads :)

You could either add an early return to `publishNewSelections()` or something like this would also work:

```suggestion
    this.subscriptions.add(
      this.unpublishedChanges
        .pipe(
          debounceTime(750),
          filter((changes) => !isEmpty(changes))
        )
        .subscribe(() => {
          this.publishNewSelections();
      })
    );
```